api/v3/Engage/Signpetition.php

```diff
@@ -193,8 +193,6 @@ function _civicrm_api3_engage_signpetition_process($params) {
     }
 
     if (!empty($activity_id)) {
-      CRM_Gpapi_Processor::updateActivityWithUTM($params, $activity_id);
```

Comment:
In `Engage.signpetition`, we still want to add UTM parameters for "real" Petition Signature activities. Removing this line would cause those to no longer be set as well.

I can see two general approaches for this:
- We only call the function for "real" Petition Signature scenarios, or
- We switch `Engage.signpetition` over to the same hook-based implementation that we'll need for `Engage.startcase`, but don't call it unless it's a "real" Petition Signature scenario (so we don't run it twice)

(Okay, it's technically the same approach, one just includes refactoring `Engage.signpetition` now rather than as a follow-up ticket. 😅 )